code/go/0chain.net/smartcontract/storagesc/blobber_alloc_partition.go

```diff
@@ -34,6 +34,9 @@ func partitionsBlobberAllocationsAdd(state state.StateContextI, blobberID, alloc
 
 	loc, err := blobAllocsParts.AddItem(state, &BlobberAllocationNode{ID: allocID})
 	if err != nil {
+		if err == partitions.ErrPartitionItemAlreadyExist {
+			return blobAllocsParts, &partitions.PartitionLocation{Location: loc}, nil
```

Comment:
The `loc` will always be `0` if the ErrPartitionItemAlreadyExist is returned.

Reply:
good catch @peterlimg 
I've fixed it